realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1070,6 +1081,7 @@ public void executeTransaction(Transaction transaction) {
      * @param transaction {@link io.realm.Realm.Transaction} to execute.
      * @param callback optional, to receive the result of this query.
      * @return a {@link RealmAsyncTask} representing a cancellable task.
+     * @throws IllegalArgumentException if the transaction is null, or if the realm is opened from another thread.
```

Comment:
`executeTransaction(Transaction,Transaction.Callback)` does not seem to have thread checking...